### ISSUE TITLE
(BUG) DIV modules in VST

### ISSUE THREAD

**Issue by Omar-Misa** [zezic/ZZC#88]
The ZZC Clock and time modulators are my favourite plugins for clocking in VCV.
i figured that they were best hope of getting song position into VCV VST.  **ZZC Clock** works great but the **DIV** time modulators aren't responding.

![clock dividers](https://user-images.githubusercontent.com/72618091/145175995-9631b565-7ce5-49f4-a7bf-36191cf1d62f.PNG)
Pictured above i have:
- Host DAW sending song position wrapping each beat and controlling **Knob 1** of **Vult Knobs**.
- **Knob 1** supplies external phase input to **ZZC Clock**.
- Phase Out of **ZZC Clock** supplies **DIV(expander)**, **DIV**, and **DIVIDER**'s phase input.
   - The purple trace on the scopes are the Phase Out of **ZZC Clock**.  The blue trace is of the **DIV**s.  As you can see, the **DIV**'s aren't responding.

![direct to clock dividers](https://user-images.githubusercontent.com/72618091/145176009-c7a87703-f3c6-4931-8fc9-500ee8649cad.PNG)
Pictured directly above i have **Vult Knobs Knob 1** directly supplying **DIV**s phase input, with the same result as before.

![24PPQ](https://user-images.githubusercontent.com/72618091/145178155-93999e57-5cc3-407f-9e03-262bffd25592.PNG)
- When i use the 24PPQ clock out put of **MIDI--CV**, the **DIV**s are responsive, weirdly.
- You'll notice, however, that **DIV** and **DIVIDER** somehow get phase input from **DIV(expander)**, instead of the connection to **ZZC Clock**, in the scenario pictured directly above.
   - i tend to avoid this configuration as there's some timing jitter when clocking to **MIDI--CV**'s Clock Outputs.

Win10 64bit
FL Studio 20.9
VCV VST 2.0.2
ZZC plugins 2.0.2

Thanks for your time and superb modules!

**Comment by zezic** [comment]
I think, there is a problem with insufficient plugin parameter update rate in FL Studio. As a workaround it should be possible to feed Rack with phase input using its audio input which is having sample-precise resolution.
I will think how I can improve it to perform correctly for your use case.

**Comment by Omar-Misa** [comment]
Thank you, zezic, for looking into this.  Using an external audio-rate phasor works well but generating it in FL Studio is a little tricky, in terms of scaling.  ZZC Clock responds well to the 24PPQ clock, relayed by the Rack VST.  The problem i previously mentioned, about ZZC Divs inheriting phase from ZZC Div(expander), seems to not be the case anymore.  Perhaps due to the update of Rack 2.0.3.

In short, all your plugins are now working perfectly.
Looking forward to Phraseque's addition to the Rack 2 library.  i intend to purchase it when it arrives.

Have a great day!
